Header header
float64[] curr_pos
